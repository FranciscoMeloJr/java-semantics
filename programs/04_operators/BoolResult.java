class main {

	void printAllB(String[] vs, boolean[] v) {
		for (int i = 0; i<v.length; i++) {
      System.out.println(vs[i]+" = "+v[i]);
		}
	}

  int nrOps = 14;

  String[] initVS() {
    String[] vs = new String[nrOps];
    int i=0;

		//Infix operators
		vs[i++] = "true || false";
		vs[i++] = "true && false";
		vs[i++] = "true | false ";
		vs[i++] = "true ^ false ";
		vs[i++] = "true & false ";
		vs[i++] = "true == false";
		vs[i++] = "true != false";
		vs[i++] = "4 == 2       ";
		vs[i++] = "4 != 2       ";
		vs[i++] = "4 < 2        ";
		vs[i++] = "4 > 2        ";
		vs[i++] = "4 <= 2       ";
		vs[i++] = "4 >= 2       ";

		//Prefix operators
		vs[i++] = "! true       ";

    return vs;
  }

	main(String[] args) {
    boolean[] vb = new boolean[nrOps];
    String[] vs = initVS();

    for(int i=0; i<nrOps; i++) {
      vb[i] = false;
    }

    int i=0;
		//Infix operators
		vb[i++] = true || false;
		vb[i++] = true && false;
		vb[i++] = true | false;
		vb[i++] = true ^ false;
    vb[i++] = true & false;
    vb[i++] = true == false;
    vb[i++] = true != false;
    vb[i++] = 4 == 2;
    vb[i++] = 4 != 2;
    vb[i++] = 4 < 2;
    vb[i++] = 4 > 2;
    vb[i++] = 4 <= 2;
    vb[i++] = 4 >= 2;

		//Prefix operators
		vb[i++] = ! true;

    printAllB(vs,vb);
    System.out.println("Done!");
	}
}

public class BoolResult {
  public static void main(String[] args) {
    new main(args);
  }
}
