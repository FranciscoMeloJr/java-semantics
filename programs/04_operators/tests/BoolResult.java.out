true || false = true
true && false = false
true | false  = true
true ^ false  = true
true & false  = false
true == false = false
true != false = true
4 == 2        = false
4 != 2        = true
4 < 2         = false
4 > 2         = true
4 <= 2        = false
4 >= 2        = true
! true        = false
Done!
